auth/service.hh

```diff
@@ -72,10 +73,12 @@ public:
 
 /// A description of a CQL command from which auth::service can tell whether or not this command could endanger
 /// internal data on which auth::service depends.
-struct command_desc {};
+struct command_desc {
+    const resource& resource; ///< Resource impacted by this command.
+};
 
 /// A predicate on command_desc that decides whether it's safe to execute.
-using protector = bool(*)(command_desc);
+using protector = std::function<bool(command_desc)>;
```

Comment:
I knew it

Reply:
I think it's fine to use function pointer until something more complex is needed, just like it's fine to use a simple `command_desc` until a complex one becomes necessary.